File: src/entity/player/PlayerHandler.java
package entity.player;

import com.raylib.Jaylib;
import com.raylib.Raylib;
import com.raylib.Jaylib.Vector2;

import util.Colors;
import util.TextUtils;

public class PlayerHandler {
	
	private static final int maxPlayers = 4;
	public static Player[] players;
	private static boolean[] connectedControllers;

	public static void Start() {

		// Make the array to store the players
		// and controller status
		players = new Player[maxPlayers];
		connectedControllers = new boolean[maxPlayers];
	}

	public static void update() {

		// Check for if a new player joins
		ConnectNewPlayers();

		// Update all the players
		for (Player player : players) {
			
			if (player == null) continue;
			player.update();

		}

	}

	public static void render() {

		// Draw all the players
		for (Player player : players) {
			
			if (player == null) continue;
			player.render();

		}

		// Show a connection prompt if there is
		// spaces left for a player to join
		if (getConnectedPLayers() < maxPlayers)
		{
			TextUtils.drawTextWithController("Press <down_x> or <down_p> to join\n" + getConnectedPLayers() + "/" + maxPlayers, new Vector2(0, 0), 35, Colors.PURPLE);
		}
	}



	private static void ConnectNewPlayers()
	{
		// Loop through all the controller slots
		for (int i = 0; i < connectedControllers.length; i++) {
			
			// Check for if there is a free slot
			if (connectedControllers[i] == true) continue;

			// Check for if there is an available
			// controller we can use.
			if (!Raylib.IsGamepadAvailable(i)) continue;

			// Check for if the controller presses
			// the start button
			// TODO: Maybe change the button
			if (!Raylib.IsGamepadButtonPressed(i, Raylib.GAMEPAD_BUTTON_RIGHT_FACE_DOWN)) continue;

			// Make a new player for that controller
			connectedControllers[i] = true;
			players[i] = new Player(i);
		}
	}

	// Get how many controllers/players are connected rn
	public static int getConnectedPLayers() {

		int connectedPlayers = 0;
		for (int i = 0; i < maxPlayers; i++) {
			
			if (connectedControllers[i]) connectedPlayers++;
		}

		return connectedPlayers;
	}
}

File: src/entity/player/PlayerHandler.cs
using System.Numerics;
using Raylib_cs;

class PlayerHandler
{
	public static Player[] Players;
	public static int PlayerCount;

	public static readonly int MaxPlayers = 4;
	private static bool[] connectedControllers;

	public static void Start()
	{
		// Store the players, and the connected
		// controllers and their indexes
		Players = new Player[MaxPlayers];
		connectedControllers = new bool[MaxPlayers];
	}

	public static void Update()
	{
		// Update all the players
		foreach (Player player in Players)
		{
			if (player == null) continue;
			player.Update();
		}
	}

	public static void Render()
	{
		// Render all the players
		foreach (Player player in Players)
		{
			if (player == null) continue;
			player.Render();
		}
	}


	// Check for if a new controller/player joins
	public static void ConnectNewPlayers()
	{
		// Loop through all the controller slots
		for (int i = 0; i < connectedControllers.Length; i++)
		{
			// Check for if there is a free slot
			if (connectedControllers[i] == true) continue;

			// Check for if there is an available
			// controller we can use.
			if (!Raylib.IsGamepadAvailable(i)) continue;

			// Check for if the controller presses
			// the start button
			// TODO: Maybe change the button
			if (!Raylib.IsGamepadButtonPressed(i, GamepadButton.RightFaceDown)) continue;

			// Make a new player for that controller
			// and increase the connected players counter
			connectedControllers[i] = true;
			Players[i] = new Player(i);
			PlayerCount++;
		}
	}
}